tools/hookgen/HookGen.cpp

```diff
@@ -214,7 +218,7 @@ HookGen::writeEventToPublicHeader(const char *name, const char *description, con
 	char *exampleBuf = (char *)malloc(bufSize + 1);
 	if (NULL != exampleBuf) {
 		/* If we fail to allocate room for the example just skip writing it */
-		int length = sprintf(exampleBuf, (char *)example, name, description, name, structName);
+		int length = snprintf(exampleBuf, bufSize + 1, (char *)example, name, description, name, structName);
 		exampleBuf[length] = '\0';
```

Comment:
In general, `snprintf()` might return `bufSize + 1 ` (or larger) if there isn't room for the NUL terminator.
In this case we're guaranteed room because `bufSize +1` is 8 more than necessary (4 copies of `%s` replaced by strings whose lengths we've accounted for). A safer approach is to say
```C
exampleBuf[bufSize] = '\0';
```
which handle the case where `snprintf()` did not terminate the result.
Alternatively, a comment explaining why `length <= bufSize` would be helpful.

Reply:
I changed the code to `exampleBuf[bufSize] = '\0';`